src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1197,49 +1211,50 @@ public void removeCertificateUser(UUID userUuid) {
 
     @Override
     @ExternalAuthorization(resource = Resource.CERTIFICATE, action = ResourceAction.LIST, parentResource = Resource.RA_PROFILE, parentAction = ResourceAction.LIST)
-    public Long statisticsCertificateCount(SecurityFilter filter) {
+    public Long statisticsCertificateCount(SecurityFilter filter, boolean includeArchived) {
         setupSecurityFilter(filter);
-        return certificateRepository.countUsingSecurityFilter(filter);
+        return certificateRepository.countUsingSecurityFilter(filter, includeArchived);
     }
 
     @Override
     @ExternalAuthorization(resource = Resource.CERTIFICATE, action = ResourceAction.LIST, parentResource = Resource.RA_PROFILE, parentAction = ResourceAction.LIST)
-    public StatisticsDto addCertificateStatistics(SecurityFilter filter, StatisticsDto dto) {
+    public StatisticsDto addCertificateStatistics(SecurityFilter filter, StatisticsDto dto, boolean includeArchived) {
         setupSecurityFilter(filter);
 
         long start = System.nanoTime();
         try (ExecutorService executor = Executors.newVirtualThreadPerTaskExecutor()) {
             executor.invokeAll(List.of(
                     () -> {
-                        dto.setCertificateStatByKeySize(certificateRepository.countGroupedUsingSecurityFilter(filter, null, Certificate_.keySize, null));
+                        dto.setCertificateStatByKeySize(certificateRepository.countGroupedUsingSecurityFilter(filter, null, Certificate_.keySize, null, includeArchived));
```

Comment:
This should be handled by adding predicate for that field to additionalWhereClause. In all calls of `countGroupedUsingSecurityFilter` in this method.